DevEdu.API/DevEdu.DAL/Repositories/GroupRepository.cs

```diff
@@ -23,6 +31,124 @@ public class GroupRepository : BaseRepository, IGroupRepository
         private const string _taskGroupSelectByIdProcedure = "dbo.Group_Task_SelectById";
         private const string _taskGroupUpdateProcedure = "dbo.Group_Task_Update";
 
+        public int AddGroup(GroupDto groupDto)
+        {
+            return _connection.QuerySingle<int>
+            (
+                _groupInsertProcedure,
+                new
+                {
+                    groupDto.Id,
+                    groupDto.Name,
+                    groupDto.Course
+                },
+                commandType: CommandType.StoredProcedure
+            );
+        }
+
+        public void DeleteGroup(int id)
+        {
+            _connection.Execute
+            (
+                _groupDeleteProcedure,
+                new { Id = id},
+                commandType: CommandType.StoredProcedure
+            );
+        }
+
+        public GroupDto GetGroup(int id)
+        {
+            var userDictionary = new Dictionary<int, UserDto>();
+            GroupDto result = default;
+            UserDto resultUser = default;
+            int index = 0;
+            return _connection
+                .Query<GroupDto, CourseDto, Role, UserDto, GroupDto>

```

Comment:
ah, here is something you've got to change
you are getting the list of users here
instead of this, you should do it in the service